rest-api/src/main/java/com/cloudbees/workflow/rest/external/StageNodeExt.java

```diff
@@ -55,23 +57,7 @@ public void setStageFlowNodes(List<AtomFlowNodeExt> stageFlowNodes) {
     }
 
     public static boolean isStageNode(FlowNode node) {
-        return (node.getAction(StageAction.class) != null);
-    }
-
-    public static StageNodeExt create(FlowNode node) {
-        StageNodeExt stageNodeExt = new StageNodeExt();
-
-        stageNodeExt.addBasicNodeData(node);
-
-        // Use the last node in the stage to configure the stage status.
-        FlowNode stageEndNode = FlowNodeUtil.getStageEndNode(node);
-        if (NotExecutedNodeAction.isExecuted(stageEndNode)) {
-            stageNodeExt.setStatus(StatusExt.valueOf(stageEndNode.getError()));
-        } else {
-            stageNodeExt.setStatus(StatusExt.NOT_EXECUTED);
-        }
-
-        return stageNodeExt;
+        return (node.getAction(LabelAction.class) != null && node.getAction(ThreadNameAction.class) == null);
```

Comment:
@oleg-nenashev What's your error case here?  We could do a type/descriptor check, but that is far more brittle in the face of subtle implementation changes and this actually lets stage view handle some other labelling schemes if added in the future.  It's also been run by @jglick and at the time he concurred that it was a reasonable approach.